### ISSUE TITLE
Rewrite checkout pickup-point selection around a shared validated selection model

### ISSUE THREAD

**Issue by bilfeldt** [smartsendio/woocommerce#50]
## Goal

Rewrite checkout pickup-point selection for v9.1 around one server-owned selection model shared by Classic Checkout and Checkout Block. This is the foundation for alternative-address search (#152) and the map (#153), and replaces the original proposal to expose the current selector HTML through partials.

The complete rewrite belongs in 9.1. The release fixes in #197 and public-contract changes in #201 land before 9.0, and the contract shipped in 9.0 remains supported through the rewrite.

## Scope

- Separate the current search query and its result list from the selected pickup-point object. The selected point carries the carrier/country context required for server validation; a point number alone is not a sufficient identity.
- Distinguish an automatic default from an explicit customer selection. Retain the existing auto-select setting, define its behavior when the address or shipping method changes, and document that behavior. Do not reintroduce a public default-selection hook as part of the rewrite without a separately justified contract.
- Keep an explicit selection when a new result list does not contain it, provided it remains valid for the selected carrier and country. This permits a workplace pickup point outside the results around the delivery address. Show the selected point separately when needed so the customer can see what will be saved.
- Revalidate or clear incompatible selections after carrier/country changes. Reject stale client requests and late search responses that would restore a selection from an obsolete checkout context.
- Resolve and validate submitted selections on the server, including after a session-cache miss. Store the same validated object on the order in both checkout implementations; never silently discard a selection that the customer was shown as accepted.
- Preserve the supported no-points/API-failure checkout behavior. Distinguish an unavailable lookup from an invalid submitted point so outage handling does not become a bypass for incompatible selections.
- Implement a shared lookup, selection and formatting service with thin Classic Checkout and Store API/Checkout Block adapters. Replace obsolete selector code once both adapters use the new model.
- Provide accessible loading, empty, error and selected states with carrier-neutral, translatable text. The normal closest-points list remains usable without search or a map.

## Public contract and compatibility

- Preserve the 9.0 hooks `smart_send_pickup_point_search_params`, `smart_send_pickup_points_found`, `smart_send_pickup_point_timeout` and `smart_send_pickup_point_label`, including documented types and semantics. The label filter produces plain text and applies consistently to each presentation.
- Result filtering and validation of an existing explicit selection are separate operations. Do not treat absence from the latest result list as proof that a selected point is invalid.
- Keep markup, CSS selectors, JavaScript state and session-cache representation internal. The original request for configurable HTML does not require public template hooks; use supported presentation options and the stable label contract instead.
- Preserve existing shipping method IDs, merchant settings, order metadata and saved pickup-point compatibility. Use the final 9.0 namespaces rather than introducing another class-name migration.
- Build on Smart Send API v1. This milestone must not depend on the API v2/OAuth work scheduled for 9.2.

## Acceptance and validation

- [ ] Classic Checkout and Checkout Block both complete orders with the same server-validated pickup point and retain the existing auto-select setting.
- [ ] Explicit selection survives a compatible search/address update even when absent from the new result list; incompatible carrier/country changes cannot reuse it.
- [ ] Cache loss, checkout recalculation, late responses, retries and refresh do not silently lose or resurrect a selection.
- [ ] Integration coverage verifies the shared model, hook contract and persistence; browser coverage exercises both checkout adapters, auto-selection on/off, carrier changes and unavailable lookups.
- [ ] Tests and documentation cover the chosen automatic-default policy and the difference between result state and selection state.
- [ ] PHP 7.4 and the supported WordPress/WooCommerce minimums remain supported; committed build assets pass freshness checks.
- [ ] Retired selector code and documentation are removed, with no new dependency on API v2.

Milestone: v9.1.0. Depends on #201. Enables #152, then #153. Target branch: `develop`.

**Comment by bilfeldt** [comment]
Superseded by #221 – the shared selection model, alternative-address search and map are now specified together in one issue against the approved design (owner decision 2026-09-20).
